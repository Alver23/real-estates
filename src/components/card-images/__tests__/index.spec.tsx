// Dependencies
import { shallow, ShallowWrapper } from 'enzyme';

// Mocks
import mocks from './mocks.json';

// Under test file
import CardImages from '../index';

describe('<CardImages />', () => {
  let component: ShallowWrapper;

  const props = {
    images: mocks,
  };

  beforeEach(() => {
    component = shallow(<CardImages {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find('.card-images-container').children()).toHaveLength(4);
    expect(component.find('span').text()).toBe('+ 2');
  });

  it('should render correctly when send two images', () => {
    const images = [...mocks];
    images.splice(0, 2);
    component.setProps({ images });
    expect(component.find('.card-images-container').children()).toHaveLength(2);
    expect(component.find('span').exists()).toBeFalsy();
  });
});
