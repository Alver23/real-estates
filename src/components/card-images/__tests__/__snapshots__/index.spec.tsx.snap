// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardImages /> should save a snapshot of the component 1`] = `
<div
  className="card-images-container card-images-container--content-3"
>
  <div
    className="image-container-1"
    key="1"
  >
    <CardImage
      alt="fake name"
      image="https://i.picsum.photos/id/619/500/500.jpg?hmac=NGtxAgAsQGvRr6cQUzK836TwMCx1OYBGwe1j37CXX70"
    />
  </div>
  <div
    className="image-container-2"
    key="2"
  >
    <CardImage
      alt="fake name 2"
      image="https://i.picsum.photos/id/851/500/500.jpg?hmac=8qtjAOEbbDU4dZvNe0SQ5dAk0iv-zHze-29LZq1Amno"
    />
  </div>
  <div
    className="image-container-3"
    key="3"
  >
    <CardImage
      alt="fake name 4"
      image="https://i.picsum.photos/id/449/600/600.jpg?hmac=PQ7w_GECRTwxZNh3wE841cH8p4kC6WTCQ4el-tcc3Xs"
    />
  </div>
  <div
    className="absolute | right-4 | z-50 | h-full | flex items-center | font-semibold | text-white text-lg"
  >
    <span>
      + 
      1
    </span>
  </div>
</div>
`;
